### ISSUE TITLE
[vim9script] Spec

### ISSUE THREAD

**Issue by errael** [vim/vim#13146]
### Steps to reproduce

In the spec
```
A class member declaration
    static this._privateMemberName: memberType
    static this.readonlyMemberName: memberType
    static public this.readwriteMemberName: memberType
```
Shouldn't have the word `this`.

There's
```
Note that constructors cannot be declared as "static",
because they always are.
```

That's not really correct; a constructor executes as a object method in the context of the newly created object. In vim9class, and other languages, they just have special invocation rules that use the class name. If they were static, they could not use "this".

### Expected behaviour

Accurate text.

### Version of Vim

9.0.1927

### Environment

ubuntu

**Comment by errael** [comment]
And should this be something like
```
static _privateClassVariableName: variableType
...
```
with variable, not member

**Comment by yegappan** [comment]
I have incorporated some of these changes in https://github.com/vim/vim/pull/13149.

**Comment by errael** [comment]
Remaining issue from this:
```
Note that constructors cannot be declared as "static",
because they always are.
```

That's not really correct; a constructor executes as a object method in the context of the newly created object. In vim9class, and other languages, they just have special invocation rules that use the class name. If they were static, they could not use "this".

**Comment by errael** [comment]
Closed by #13756
